docs/RFCS/beta_and_experimental_features.md

```diff
@@ -0,0 +1,220 @@
+  competing product, then treat the feature as experimental. Check
+  that the differences with the competing product(s) are properly
+  documented in the RFC that introduces the new feature, and later in
+  online docs.
+
+- If it does perform very similarly, then:
+
+  - if the change is "small" as defined in the contributor guide,
+	and the feature mirrors 1-to-1 a SQL standard or a competing product,
+	then consider the feature as non-experimental.
+
+  - otherwise, consider the feature as
+	experimental, unless the RFC that introduces the new feature makes a
+	successful argument otherwise.
+
+# Reference-level explanation
+
+## Detailed design
+
+- To prepare for the application of this RFC the session variable mechanism
+  (`pkg/sql/vars.go`) must be extended to support a "description" field, like
+  cluster settings already have.
+
+- Other features currently experimental at the time of this writing must be
+  reviewed and possibly documented as such in all their UX surface area
+  (e.g. IMPORT CSV, the experimental SQL statements, maybe
+  others. Also see unresolved questions below.)
+
+- The RFC on SQL syntax changes must be updated accordingly.
+
+No other technical changes needed.
+
+## Drawbacks
+
+None known.
+
+## Rationale and Alternatives
+
+The motivation section outlines the reason why "do nothing" is not a
+viable alternative: we do want to commit to backward compatibility and
+*also* want to keep some wiggle room and release new features before
+their UX is finalized; so we must find a way to carve out new features
+away from the scope of backward compatibility.
+
+Some alternatives that were considered:
+
+- design and implement the feature "as finished" and only gate it behind a
+  `experimental` cluster setting. Problems with this approach:
+
+  - some features are not amenable to cluster settings, e.g. flags
+	that control the behavior of `cockroach start` or `cockroach sql` when
+	the server is not active yet;
+
+  - the DBA may know about the experimental nature of the feature,
+	because they set the cluster setting, but then developers with
+	access to the database may start discovering the feature after
+	it's enabled and not understand that the feature is experimental,
+	because nothing else would tell them.
+
+- design and implement the feature "as finished" and only document it
+  in the online docs. Problems with this approach:
+
+  - users may discover features without reading the docs and develop a
+	wrong expectation.
+
+- use the word `experimental` in the *name* of the flags / settings instead
+  of their description. Problems with this approach:
+
+  - if the feature evolves beyond the experimental phase to "stable"
+	without change, the client apps / deployments must still be
+	updated to use the new name. We want a path to promote a feature
+	to stable without disruption for early users.
+
+## Unresolved questions
+
+- Is it reasonable to keep the terms "Beta" and "Experimental"
```

Comment:
I find a bit odd that we use alpha and beta as release terms but beta and experimental here. I personally think alpha conveys that something is very "early".